.husky/commit-msg

```diff
@@ -1,10 +1,23 @@
 #!/bin/sh
-if ! head -1 "$1" | grep -qE "^(feat|fix|ci|chore|docs|test|style|refactor|perf|build|revert|i18n)(\(.+?\))?: .{1,}$"; then
+
+commit_msg_file=$1
+commit_msg=`cat $commit_msg_file`
+
+if [ -z "$commit_msg" ]; then
+    echo "Aborting commit. Your commit message is empty." >&2
+    exit 1
+fi
+
+first_line=`echo "$commit_msg" | head -1`
+
+if ! echo "$first_line" | grep -qE "^(feat|fix|ci|chore|docs|test|style|refactor|perf|build|revert|i18n)(\(.+?\))?: .{1,}$" && \
+   ! echo "$first_line" | grep -qE "^Merge (branch|pull request) '.+' into .+$"; then
     echo "Aborting commit. Your commit message is invalid." >&2
-    echo "It should start with: feat|fix|ci|chore|docs|test|style|refactor|perf|build|revert|i18n" >&2
+    echo "It should start with: feat|fix|ci|chore|docs|test|style|refactor|perf|build|revert|i18n, or be a merge commit." >&2
```

Comment:
@chybisov  My bad here, I noticed that it did not work the way I had thought and wasn´t as easy as just adding it to the other naming conventions. 

So, this code now actually covers those Merge messages.

Reply:
I see you are using some deep magic here 😂

Reply:
Haha, indeed! I have asked dr. GPT on that one!